packages/app/src/store/practice/mutations.ts

```diff
@@ -85,9 +64,15 @@ export const mutations: MutationTree<PracticeState> = {
     state.practiceLastQuestionCorrect = practiceLastQuestionCorrect;
   },
   addPracticeAttemptedQuestion(state: PracticeState, payload: QuestionReport) {
-    state.practiceAttemptedQuestions.questions.push(payload)
-  },
-  createQuestionReport(state: PracticeState) {
-    
+    var added = false;
```

Comment:
Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find for more concise code